errors/src/commonMain/kotlin/dev/icerock/moko/errors/mappers/ExceptionMappersStorage.kt

```diff
@@ -63,45 +71,93 @@ object ExceptionMappersStorage {
+
+        return if (mapper == null && throwable !is Exception) {
+            throw throwable
+        } else {
+            mapper
+        }
     }
 
     /**
-     * Tries to find mapper (E) -> T. First, a mapper with a condition is looked for. If this was
-     * not found, then a simple mapper is looked for.
+     * Tries to find mapper (E) -> T by [throwable] instance. First, a mapper with condition is
+     * looked for. If mapper with condition was not found, then a simple mapper is looked for. If
+     * the mapper was not found, it will return null.
+     * If there is no mapper for the [throwable] of class [E] and [E] does't inherits
+     * [kotlin.Exception], then exception will be rethrown.
      */
-    inline fun <E : BasicException, reified T : Any> find(exception: E): ((E) -> T)? =
-        find(
-            T::class,
-            exception,
-            exception::class
-        )
+    inline fun <E : Throwable, reified T : Any> find(throwable: E): ((E) -> T)? = find(
+        resultClass = T::class,
+        throwable = throwable,
+        exceptionClass = throwable::class
+    )
 
-    fun setUnknownErrorText(text: StringDesc): ExceptionMappersStorage {
-        unknownErrorText = text
-        return this
+    /**
+     * Sets fallback (default) value for [T] errors type.
+     */
+    fun <T : Any> setFallbackValue(clazz: KClass<T>, value: T): ExceptionMappersStorage {
+        fallbackValuesMap[clazz] = value
+        return ExceptionMappersStorage
+    }
+
+    /**
+     * Sets fallback (default) value for [T] errors type.
+     */
+    inline fun <reified T : Any> setFallbackValue(value: T): ExceptionMappersStorage =
+        setFallbackValue(T::class, value)
+
+    /**
+     * Returns fallback (default) value for [T] errors type.
+     * If there is no default value for the class [T], then [FallbackValueNotFoundException]
+     * exception will be thrown.
+     */
+    fun <T : Any> getFallbackValue(clazz: KClass<T>): T {
+        return fallbackValuesMap[clazz] as? T
+            ?: throw FallbackValueNotFoundException(clazz)
     }
-}
 
-fun <E : BasicException> ExceptionMappersStorage.throwableToStringDesc(e: E): StringDesc {
-    return find<E, StringDesc>(e)?.invoke(e) ?: unknownErrorText
+    /**
+     * Returns fallback (default) value for [T] errors type.
+     * If there is no default value for the class [T], then [FallbackValueNotFoundException]
+     * exception will be thrown.
+     */
+    inline fun <reified T : Any> getFallbackValue(): T = getFallbackValue(T::class)
+
+    /**
+     * Factory method that creates mappers (Throwable) -> T with a registered fallback value for
+     * class [T].
+     */
+    inline fun <E : Throwable, reified T : Any> throwableMapper(): (e: E) -> T {
+        val clazz = T::class
+        val fallback = getFallbackValue(clazz)
+        return { e ->
+            find(clazz, e, e::class)?.invoke(e) ?: fallback
+        }
+    }
```

Comment:
you should not inline this. every place where you call this method will be replace to this code. this is overhead.
rollback please old solution with 2 methods :)

Reply:
Fixed. But I think everyone will use the inline function :)